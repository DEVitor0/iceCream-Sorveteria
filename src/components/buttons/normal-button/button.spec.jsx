import React from 'react';
import { render, screen } from '@testing-library/react';
import NormalButton from './index';
import PropTypes from 'prop-types';

test('renders button and displays children correctly', () => {
  const { asFragment } = render(<NormalButton>Click Me</NormalButton>);

  const buttonElement = screen.getByText(/Click Me/i);
  expect(buttonElement).toBeInTheDocument();

  expect(asFragment()).toMatchSnapshot();
});

test('renders button with children component (Anchor) correctly', () => {
  const AnchorLink = ({ href, text }) => <a href={href}>{text}</a>;

  AnchorLink.propTypes = {
    href: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
  };

  const { asFragment } = render(
    <NormalButton>
      <AnchorLink href="https://test.com.br" text="Go to Example" />
    </NormalButton>,
  );

  const linkElement = screen.getByText(/Go to Example/i);
  expect(linkElement).toBeInTheDocument();

  expect(asFragment()).toMatchSnapshot();
});

test('button has correct class for styling', () => {
  const { asFragment } = render(<NormalButton>Click Me</NormalButton>);

  const buttonElement = screen.getByText(/Click Me/i).closest('button');
  expect(buttonElement).toHaveClass('normal-button');

  expect(asFragment()).toMatchSnapshot();
});
