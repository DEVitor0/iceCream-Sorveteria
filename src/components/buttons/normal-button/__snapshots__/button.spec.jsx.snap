// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button has correct class for styling 1`] = `
<DocumentFragment>
  <button
    class="normal-button"
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`renders button and displays children correctly 1`] = `
<DocumentFragment>
  <button
    class="normal-button"
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`renders button with children component (Anchor) correctly 1`] = `
<DocumentFragment>
  <button
    class="normal-button"
  >
    <a
      href="https://test.com.br"
    >
      Go to Example
    </a>
  </button>
</DocumentFragment>
`;
